### ISSUE TITLE
Reflective XSS Vulnerability in Svelte Site

### ISSUE THREAD

**Issue by pau101** [sveltejs/kit#1291]
The site sub-project is vulnerable to reflective XSS via the defined [error handler](https://github.com/sveltejs/svelte/blob/c24b313b1af8c55db4fe0ce1ea689000042a2531/site/src/server.js#L10-L14) response data including unescaped javascript error messages from middleware, which may contain unsanitized user input.

One route that enables an attacker to cause such an uncaught error is [`/repl/[id].json`](https://github.com/sveltejs/svelte/blob/c24b313b1af8c55db4fe0ce1ea689000042a2531/site/src/routes/repl/%5Bid%5D/index.json.js). Due to insufficient error handling a malformed id passed to the [gist database query](https://github.com/sveltejs/svelte/blob/c24b313b1af8c55db4fe0ce1ea689000042a2531/site/src/routes/repl/%5Bid%5D/index.json.js#L88-L92) throws and propagates with a message that includes the invalid and unsanitized gist id as specified by the path. The error generated has a message of the form:

    invalid input syntax for type uuid: "{id}"`
When this message is emitted as response data the browser happily understands `{id}` as any html, executing whatever scripts. 

This issue can be observed by browsing https://svelte.dev/repl/%3Cbody%20onload=alert(1)%3E.json, which produces the resulting page, which when loaded simply prompts an alert:

```html
 invalid input syntax for type uuid: "<body onload=alert(1)>"
```

**Comment by Rich-Harris** [comment]
Is there a way to reproduce this in SvelteKit?

**Comment by Conduitry** [comment]
@Rich-Harris Yep, if a endpoint handler function throws an exception (instead of returning a proper response), the message of the exception will be used in the response from the server, with an HTML mime type. If the error message contains the URL (or some other user-supplied text) this can potentially be used as part of an XSS attack.

Ideally, endpoint handling functions would return something rather than throwing an exception, but if they do, the framework should probably display that with a text MIME type to be safe.

**Comment by Conduitry** [comment]
@Rich-Harris I just tried again I don't seem to be able to reproduce this anymore - it's possible that it was fixed in the past week or so. When I throw an error from an endpoint, the error's message is no longer returned as part of a `text/html` response, whether during dev or preview or with the Node adapter. There's now no `Content-Type` header at all, and so it's interpreted as plain text, and so that's not a possible XSS attack vector. Before we close this, I'd like to confirm that I'm indeed doing the same thing that I was doing before when I was reproducing this with an older version - and then it may also be a good idea to add a test for this.
